src/wp-includes/class-wp-modules.php

```diff
@@ -0,0 +1,245 @@
+<?php
+/**
+ * Modules API: WP_Modules class.
+ *
+ * Native support for ES Modules and Import Maps.
+ *
+ * @package WordPress
+ * @subpackage Modules
+ */
+
+/**
+ * Core class used to register modules.
+ */
+class WP_Modules {
+	/**
+	 * Holds the registered modules, keyed by module identifier.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @var array
+	 */
+	private static $registered = array();
+
+	/**
+	 * Holds the module identifiers that were enqueued before registered.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @var array
+	 */
+	private static $enqueued_before_registered = array();
```

Comment:
Nit-pick: Property docs typically shouldn't have a blank line between those annotations.
```suggestion
	/**
	 * Holds the registered modules, keyed by module identifier.
	 *
	 * @since 6.5.0
	 * @var array
	 */
	private static $registered = array();

	/**
	 * Holds the module identifiers that were enqueued before registered.
	 *
	 * @since 6.5.0
	 * @var array
	 */
	private static $enqueued_before_registered = array();
```

Reply:
Done in https://github.com/WordPress/wordpress-develop/pull/5818/commits/e30c8e9ae0849f863bc730c131bd594edfc38ef8.